gobblin-yarn/src/main/java/org/apache/gobblin/yarn/AbstractYarnAppSecurityManager.java

```diff
@@ -166,6 +175,15 @@ public void loginAndScheduleTokenRenewal() {
     }
   }
 
+  protected void sendTokenFileUpdatedMessage() {
+    // Send a message to the controller (when the cluster is not managed)
+    // and all the participants if this is not the first login
+    if (!this.isHelixClusterManaged) {
```

Comment:
Why do we only send token file update message to controller when it's not managed? Will helix maintain the token refresh for us?

Reply:
In the managed mode, the GobblinClusterManager joins as a PARTICIPANT and hence, will get the token file updated message and handle the refresh. In the managed mode, sending this message to the controller will only queue messages on the CONTROLLER, since the CONTROLLER will likely not have a handler. In the unmanaged cluster mode, we maintain the same behavior as before, sending the updated message to both CONTROLLER and PARTICIPANT.